src/components/views/rooms/PresenceLabel.tsx

```diff
@@ -37,38 +38,14 @@ export default class PresenceLabel extends React.Component<IProps> {
         activeAgo: -1,
     };
 
-    // Return duration as a string using appropriate time units
-    // XXX: This would be better handled using a culture-aware library, but we don't use one yet.
-    private getDuration(time: number): string | undefined {
-        if (!time) return;
-        const t = Math.round(time / 1000);
-        const s = t % 60;
-        const m = Math.round(t / 60) % 60;
-        const h = Math.round(t / (60 * 60)) % 24;
-        const d = Math.round(t / (60 * 60 * 24));
-        if (t < 60) {
-            if (t < 0) {
-                return _t("%(duration)ss", { duration: 0 });
-            }
-            return _t("%(duration)ss", { duration: s });
-        }
-        if (t < 60 * 60) {
-            return _t("%(duration)sm", { duration: m });
-        }
-        if (t < 24 * 60 * 60) {
-            return _t("%(duration)sh", { duration: h });
-        }
-        return _t("%(duration)sd", { duration: d });
-    }
-
     private getPrettyPresence(presence?: string, activeAgo?: number, currentlyActive?: boolean): string {
         // for busy presence, we ignore the 'currentlyActive' flag: they're busy whether
         // they're active or not. It can be set while the user is active in which case
         // the 'active ago' ends up being 0.
         if (presence && BUSY_PRESENCE_NAME.matches(presence)) return _t("Busy");
 
         if (!currentlyActive && activeAgo !== undefined && activeAgo > 0) {
-            const duration = this.getDuration(activeAgo);
+            const duration = formatDuration(activeAgo);
```

Comment:
:rocket: , though there's an argument it should have been a separate PR: it's quite a different class of change to the rest of the changes here.

Reply:
don't worry now we're here. One for next time.